mobly/controllers/android_device.py

```diff
@@ -897,19 +897,22 @@ def unload_snippet(self, name):
 
   def generate_filename(self,
                         file_type,
+                        filename=None,
```

Comment:
I don't think you actually need to add this?
you can simply use `file_type` to do what you need?

Reply:
May user confuse for the filename showing on the file_type?

Reply:
The "user" here is `AndroidDevice`'s own `take_screenshot`, so I don't think we should worry about it?

Reply:
According to [go/mobly-ad-output-files](http://go/mobly-ad-output-files),
the file type naming convention is designed to make accessing generated files easily.
Please consider to change 'debug_tag' or override 'generate_filename' in your test class to have special, fully customized file name.

Reply:
Give the filename to screenshot directly, not modify the 'generate_filename' funtion